### PR TITLE
Minor: Make execute_input_stream Accessible for Any Sinking Operators

### PR DESCRIPTION
**Pull request by berkaysynnada**
## Which issue does this PR close?

Closes #.

## Rationale for this change

Currently, DF does not utilize `execute_input_stream()` outside of the `DataSinkExec` implementation. However, many downstream users might benefit from such a utility when writing their own sink operators. Therefore, moving it to a broader scope could be helpful.

## What changes are included in this PR?

The `execute_input_stream()` function has been moved from the `DataSinkExec` implementation to `lib.rs` in the `physical_plan` module.

### CODE REVIEW

**Review by ozankabak — APPROVED** [review]
LGTM thank you

File: datafusion/physical-plan/src/lib.rs [hunk 1]
```diff
@@ -805,6 +805,97 @@ pub fn execute_stream_partitioned(
     Ok(streams)
 }
 
+/// Executes an input stream and ensures that the resulting stream adheres to
```
Comment (alamb) [inline datafusion/physical-plan/src/lib.rs:808]: ❤️  for the documentation

**Review by alamb — APPROVED** [review]
Looks good to me. Thanks @berkaysynnada  and @ozankabak  for the review

BTW one thing that we might consider based on @comphead 's discussion https://github.com/apache/datafusion/issues/11375 is if we should put these functions in a module like `util`: `datafusion/physical-plan/src/utils.rs`

However nothing else in this crate does so and thus I think this PR follows the pattern of the existing code and so 👍  from me

cc @devinjdangelo

**Comment by alamb** [comment]
Let's move the code around in a subsequent PR if we think that is useful

**Comment by alamb** [comment]
Thanks again @berkaysynnada and @ozankabak

**Pull request merged**